cmd/peer-rpc-server.go

```diff
@@ -397,5 +398,66 @@ func registerPeerRPCRouter(router *mux.Router) {
 	rpcServer, err := NewPeerRPCServer()
 	logger.FatalIf(err, "Unable to initialize peer RPC Server")
 	subrouter := router.PathPrefix(minioReservedBucketPath).Subrouter()
-	subrouter.Path(peerServiceSubPath).HandlerFunc(httpTraceHdrs(rpcServer.ServeHTTP))
+	subrouter.Path(peerServiceSubPath).HandlerFunc((rpcServer.ServeHTTP))
+}
+
+// ListenTraceNotificationArgs - listen trace notification RPC arguments.
+type ListenTraceNotificationArgs struct {
+	AuthArgs `json:"-"`
+	TargetID trace.TargetID `json:"targetId"`
+	Addr     xnet.Host      `json:"addr"`
+}
+
+// ListenTraceNotification - handles listen bucket notification RPC call.
+// It creates PeerRPCClient target which pushes requested events to target in remote peer.
+func (receiver *peerRPCReceiver) ListenTraceNotification(args *ListenTraceNotificationArgs, reply *VoidReply) error {
```

Comment:
One problem of doing rpc for trace over the network might be very chatty, not sure if it might impact the server throughput. We may need to say if trace is enabled it may perhaps reduce the cluster overall performance.

Reply:
that's true

Reply:
We should discuss this..